### PR TITLE
Chore: Move NotificationPreferences to new Preference Framework.

### PR DESCRIPTION
**Pull request by cbuchli**
Chore: Move NotificationPreferences to new Preference Framework.
Test: Switch to EqualsWithDelta for all timestamp asserts

### CODE REVIEW

File: src/Classes/Model/Execution.php [hunk 1]
```diff
@@ -99,6 +99,11 @@ class Execution extends AbstractModel
      */
     protected $latestHeartbeat;
 
+    /**
+     * @var bool
+     */
+    protected $autoExecuted = false;
```
Comment (mweibel) [inline src/Classes/Model/Execution.php]: this doesn't need to be saved in the DB?

File: src/Classes/Model/Preferences/UserNotifications.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+namespace Helio\Panel\Model\Preferences;
+
+class UserNotifications implements \JsonSerializable
+{
+    const DEFAULT_MUTE_ADMIN = false;
+    const DEFAULT_EMAIL_ON_JOB_READY = true;
+    const DEFAULT_EMAIL_ON_JOB_DELETED = false;
+    const DEFAULT_EMAIL_ON_EXECUTION_STARTED = false;
+    const DEFAULT_EMAIL_ON_EXECUTION_ENDED = true;
+    const DEFAULT_EMAIL_ON_AUTOSCHEDULED_EXECUTION_ENDED = true;
```
Comment (mweibel) [inline src/Classes/Model/Preferences/UserNotifications.php]: `private const` maybe, unsure about it tho.

File: src/Classes/Model/Execution.php (hunk 1 above)
Comment (cbuchli) [reply src/Classes/Model/Execution.php]: of course! thanks

File: src/Classes/Model/Preferences/UserNotifications.php (hunk 2 above)
Comment (cbuchli) [reply src/Classes/Model/Preferences/UserNotifications.php]: yeah, does make sense

File: src/Classes/Model/Execution.php (hunk 1 above)
Comment (mweibel) [reply src/Classes/Model/Execution.php]: hmm, `type` not needed?

**Pull request merged**